lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1774,6 +1752,7 @@ def _on_pebble_ready(self, _: WorkloadEvent):
     def _on_relation_created(self, _: RelationCreatedEvent) -> None:
         """Event handler for `relation_created`."""
         if not self._container.can_connect():
+            logger.debug("Cannot connect to container to check promtail!")
```

Comment:
Why do we need to log if we coded the library so that this will occur on pebble_ready?

Reply:
Did we though? That should be opened as a new issue, since we don't do this there.

* `_on_pebble_ready` -> `_setup_promtail` -> `_create_directories` (no guard around connect)
* Then `_container.push` (no guard)
* ...
* `_obtain_promtail` -> `_push_binary_to_workload` (no guard)
* `self._container.restart` (no guard)

This is possibly the only place we would ever log a message of any kind about "hey, this is why there's no promtail running".